Makefile

```diff
@@ -92,6 +92,8 @@ checksum: FORCE
 diffconfig: FORCE
 	mkdir -p $(BIN_DIR)
 	$(SCRIPT_DIR)/diffconfig.sh > $(BIN_DIR)/config.seed
+	echo "# buildroot $$(git --git-dir $(TOPDIR)/.git rev-parse HEAD)" > $(BIN_DIR)/feeds.seed
```

Comment:
Note that this may fail if buildroot does not reside in Git (which is something we do support at least for releases). This can for example happen if you download a release tarball from the Github release page - such a tarball will come without any metadata.

Maybe it would be better to use the output of `./scripts/getver.sh` instead which is able to cope with such a situation.

Reply:
From my POV it's not `diffconfig` target anymore, you're as well adding feeds config and build config which are probably separate things.

So this probably should be something like:

```
buildconfig: FORCE
    echo "commit=$$(git --git-dir $(TOPDIR)/.git rev-parse HEAD)" > $(BIN_DIR)/build.seed

feedsconfig: FORCE
   $(SCRIPT_DIR)/feeds list -fs >> $(BIN_DIR)/feeds.seed   

prepare: .config $(tools/stamp-compile) $(toolchain/stamp-compile)
	$(_SINGLE)$(SUBMAKE) -r diffconfig feedsconfig buildconfig
```

so you can then just  source `build.seed` in some script/make target and simply use it, without any additional parsing. And I think, that it's likely, that we (or downstream projects) might need to add also other variables to the build.seed in order to provide enough information for reproducible builds.

Reply:
Switched to using `getver.sh` and introduced `buildconfig` and `feedsconfig`.